weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/tsDataModelHooksEvaluationComparison.ts

```diff
@@ -466,14 +485,69 @@ const fetchEvaluationComparisonResults = async (
   const evalTraceIds = Object.values(summaryData.evaluationCalls).map(
     call => call.traceId
   );
+
   // First, get all the children of the evaluations (predictAndScoreCalls + summary)
   const evalTraceResProm = traceServerClient
     .callsStreamQuery({
       project_id: projectId,
       filter: {trace_ids: evalTraceIds, parent_ids: evaluationCallIds},
     })
     .then(predictAndScoreCallRes => {
-      // Then, get all the children of those calls (predictions + scores)
+      // Check for imperative evaluations and handle specially
+      const hasImperativeEvals = predictAndScoreCallRes.calls.some(
+        call =>
+          call.op_name.includes('Evaluation.predict_and_score:') ||
```

Comment:
why is this the check? a) this seems like it would match non imperatives, but also seems very fragile

Reply:
I wonder if we named all these things `EvaluationLogger.` instead of `Evaluation.` it would make this easier in a number of locations